pennylane/beta/tapes/qnode.py

```diff
@@ -0,0 +1,541 @@
+        # that allows controlled rotations to return the correct gradients.
+        #
+        # if isinstance(self.qtape, QubitParamShiftTape):
+        #     # controlled rotations aren't supported by the parameter-shift rule
+        #     stop_at = set(self.device.operations) - {"CRX", "CRZ", "CRY", "CRot"}
+
+        # expand out the tape, if any operations are not supported on the device
+        if not {op.name for op in self.qtape.operations}.issubset(stop_at):
+            self.qtape = self.qtape.expand(
+                depth=self.max_expansion, stop_at=lambda obj: obj.name in stop_at
+            )
+
+    def __call__(self, *args, **kwargs):
+        # construct the tape
+        self.construct(args, kwargs)
+
+        # execute the tape
+        return self.qtape.execute(device=self.device)
+
+    def to_tf(self, dtype=None):
+        """Apply the TensorFlow interface to the internal quantum tape.
+
+        Args:
+            dtype (tf.dtype): The dtype that the TensorFlow QNode should
+                output. If not provided, the default is ``tf.float64``.
+
+        Raises:
+            QuantumFunctionError: if TensorFlow >= 2.1 is not installed
+        """
+        # pylint: disable=import-outside-toplevel
+        try:
+            import tensorflow as tf
+            from pennylane.beta.interfaces.tf import TFInterface
+
+            self.interface = "tf"
+
+            if not isinstance(self.dtype, tf.DType):
+                self.dtype = None
+
+            self.dtype = dtype or self.dtype or TFInterface.dtype
+
+            if self.qtape is not None:
+                TFInterface.apply(self.qtape, dtype=self.dtype)
+
+        except ImportError:
+            raise QuantumFunctionError(
+                "TensorFlow not found. Please install the latest "
+                "version of TensorFlow to enable the 'tf' interface."
+            )
+
+    def to_torch(self, dtype=None):
+        """Apply the Torch interface to the internal quantum tape.
+
+        Args:
+            dtype (tf.dtype): The dtype that the Torch QNode should
+                output. If not provided, the default is ``torch.float64``.
+
+        Raises:
+            QuantumFunctionError: if PyTorch >= 1.3 is not installed
+        """
+        # pylint: disable=import-outside-toplevel
+        try:
+            import torch
+            from pennylane.beta.interfaces.torch import TorchInterface
+
+            self.interface = "torch"
+
+            if not isinstance(self.dtype, torch.dtype):
+                self.dtype = None
+
+            self.dtype = dtype or self.dtype or TorchInterface.dtype
+
+            if self.qtape is not None:
+                TorchInterface.apply(self.qtape, dtype=self.dtype)
+
+        except ImportError:
+            raise QuantumFunctionError(
+                "PyTorch not found. Please install the latest "
+                "version of PyTorch to enable the 'torch' interface."
+            )
+
+    def to_autograd(self):

```

Comment:
Why do we skip the `dtype` argument for this interface?

Reply:
Oops. I'll add it back in.

Reply:
Oh, I just had a look. The Autograd interface doesn't take a dtype argument 😆

Reply:
I guess it just follows the dtype of the numpy array?